### ISSUE TITLE
Some elements are not updated in the page

### ISSUE THREAD

**Issue by simulot** [maxence-charriere/go-app#530]
Sorry to disturb again.
Thanks for your  work. It's a real gain to share code between back and front end, and avoid JS oddities. 
 
This time I can't explain the behavior of a component I wrote. 
The context is a setting screen with few groups of options sharing the same validation logic. A component manages the group and provide a feedback to be displayed below the input field. 

I can't get it working even with all fields exportable. During tests, I have seen a behavior that I can't understand.  

The field1 has the Help text initialized with a constant. The component should update the help text on each change, but it doesn't.
The field2 component is the same, except that Help text is initialized with the same value as input field. On change, the help text is updated with field value, not with help text.  

You can check it visually: 
https://user-images.githubusercontent.com/5980377/117777672-1df0c680-b23d-11eb-9482-d2c9e90c6222.mp4

You will find code that illustrate the point here https://github.com/simulot/test-go-app.
May be I still miss the point.

**Comment by maxence-charriere** [comment]
When your on change is called, it trigger a Emit operation which adds Page, Field1 and Field2 in the update pipeline.

When Page is updated, Render() is called and a new version fo field 1 is compared with the current one:
|Component|Field|current|new|Update?|
|---|---|---|---|---|
|Page|Field1|`Initial value field1`|`No changes?`|Yes because exported field is different|
|Page|Field2|`Initial value field2`|`Same as field?`|Yes because exported field is different|
|Field1|Help|`No changes?`|`All good`|Yes because exported field is different|
|Field2|Help|`Same as field?`|`Same as field?`|No because exported field are same different|

Also you don't need this since it is overridden by field values in Page:
```go
func (f *field2) valueChanged(ctx app.Context, e app.Event) {
	f.Value = ctx.JSSrc().Get("value").String()
	if f.ReportChange != nil {
		v, err := f.ReportChange(f.Value)
		if err == nil {
			f.HelpClass = "green"  // Color is not working because overridden by "" in page
		} else {
			f.HelpClass = "red"    // Color is not working because overridden by "" in page
		}
		f.Help = v                     // <- This is overriden in Page
		log.Printf("ReportChange Help: %s, Class:%s", f.Help, f.HelpClass)
	}
}
```

If you want those to work as expected, you must have your `HelpClass`, `Help` non exported.

See it like this:
- Exported = Attribute
- Nonexported = State

**Comment by simulot** [comment]
Thanks,
Making help and helpClass not exportable has fixed the issue.
The fix for my last issue #524 also for an update problem was to make fields exportable.

**Comment by maxence-charriere** [comment]
Closing this. Reopen if you need :)
